adoc-mode.el

```diff
@@ -2148,11 +2149,9 @@ Use this function as matching function MATCHER in `font-lock-keywords'."
 (defun adoc-unfontify-region-function (beg end)
   (font-lock-default-unfontify-region beg end)
 
-  ;; remove overlays. Currently only used by AsciiDoc replacements
-  ;; TODO: this is an extremely brute force solution and interacts very badly
-  ;; with many (minor) modes using overlays such as flyspell or ediff
-  (when adoc-insert-replacement
-    (remove-overlays beg end))
+  (cl-loop for ol being the overlays from beg to end
```

Comment:
I'm not a big fan of `cl-loop` (as it tends to confuse people not familiar with CL), so I'd suggest replacing it with a different looping construct.